HVSample/src/main/java/com/microsoft/hsg/android/hvsample/PersonalImageLoader.java

```diff
@@ -33,80 +34,94 @@
 
 public class PersonalImageLoader implements ComponentCallbacks2 {
 
-	private ImageLruCache cache;
-	private HealthVaultClient hvClient;
-	private Activity context;
-	
-	private ImageView imageView;
-	
+	private ImageLruCache mCache;
+	private HealthVaultClient mClient;
+	//private Activity mContext;
+	private WeakReference<Activity> mContext;
+
 	public PersonalImageLoader(Activity context,
 			HealthVaultClient client) {
 		ActivityManager am = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
-        int memSize = am.getMemoryClass() * 1024 * 1024;
-        
-        cache = new ImageLruCache(memSize);
-        hvClient = client;
-        this.context = context;
+		int memSize = am.getMemoryClass() * 1024 * 1024;
+
+		mCache = new ImageLruCache(memSize);
+		mClient = client;
+		mContext = new WeakReference<Activity>(context);
 	}
 
 	public void load(String id, ImageView imageView, int defaultresource) {
 		imageView.setImageResource(defaultresource);
-        Bitmap image = cache.get(id);
-        if (image != null) {
-            imageView.setImageBitmap(image);
-        }
-        else {
-        	List<Record> records = HealthVaultApp.getInstance().getRecordList();
-        	Record record = null;
-        	
-        	for(Record rcd : records) {
-        		if(id == rcd.getId()) {
-        			record = rcd;
-        			break;
-        		}
-        	}
-        	
-        	hvClient.asyncRequest(getImageAsync(record), new PersonalImageCallback(id, imageView));
-        }
+		Bitmap image = mCache.get(id);
+		if (image != null) {
+			imageView.setImageBitmap(image);
+		} else {
+			List<Record> records = HealthVaultApp.getInstance().getRecordList();
+			Record record = null;
+
+			for(Record rcd : records) {
+				if(id == rcd.getId()) {
+					record = rcd;
+					break;
+				}
+			}
+
+			mClient.asyncRequest(getImageAsync(record), new PersonalImageCallback(id, imageView));
+		}
 	}
 	
 	private Callable<Bitmap> getImageAsync(final Record record) {
 		return new Callable<Bitmap>() {
-    		public Bitmap call() throws URISyntaxException, IOException {
-    			// check if it exist in file
-    			File cacheDir = context.getCacheDir();
-    			
-    			if(!cacheDir.exists()) {
-    				cacheDir.mkdirs();
-    			}
-    			
-    			File file = new File(cacheDir, "personalimage" + record.getId());
-    			
-    			if(file.exists()) {
-    				FileInputStream in = new FileInputStream(file);
-    				return BitmapFactory.decodeStream(in);
-    			}
-    			
-    			// try to get from web
-    			ThingResponseGroup2 response = record.getThings(ThingRequestGroup2.thingTypeQuery(PersonalImage.ThingType));
-    			List<Thing2> things = response.getThing();
-    			
-    			if(things != null && !things.isEmpty()) {
-    				Thing2 thing = things.get(0);
+			public Bitmap call() throws URISyntaxException, IOException {
+				// check if it exist in file
+				File cacheDir = mContext.get().getCacheDir();
```

Comment:
mContext is a weakreference now so it can be null upon get() call. Need to null check before using.